src/cls/lock/cls_lock_client.h

```diff
@@ -87,11 +89,31 @@ namespace rados {
 	void set_tag(const std::string& t) { tag = t; }
 	void set_description(const std::string& desc) { description = desc; }
 	void set_duration(const utime_t& e) { duration = e; }
-	void set_renew(bool renew) {
+
+	template<typename R, typename P>
+	void set_duration(const std::chrono::duration<R,P>& d) {
+	  std::chrono::seconds secs =
+	    std::chrono::duration_cast<std::chrono::seconds>(d);
+	  std::chrono::nanoseconds nanosecs =
+	    std::chrono::duration_cast<std::chrono::nanoseconds>(d);
+	  nanosecs -= secs;
+	  duration = utime_t(secs.count(), nanosecs.count());
```

Comment:
there's a utime_t constructor that takes ceph::real_time/coarse_real_time. how about this?
```c++
#include "common/ceph_time.h"
...
void set_duration(const ceph::timespan& d) {
  duration = utime_t(ceph::real_clock::zero() + d);
}
```

Reply:
adam has always been there before us